x-pack/plugins/security_solution/server/fleet_integration/handlers/create_default_policy.ts

```diff
@@ -48,10 +49,13 @@ export const createDefaultPolicy = (
     cloud?.isServerlessEnabled
   );
 
-  let defaultPolicyPerType: PolicyConfig =
-    config?.type === 'cloud'
-      ? getCloudPolicyConfig(factoryPolicy)
-      : getEndpointPolicyWithIntegrationConfig(factoryPolicy, config);
+  // config won't be defined if inputs is an empty array, this happens on managed cloud onboarding.
+  const useCloudConfig =
+    config?.type === 'cloud' || (!config && cloud?.isCloudEnabled && !cloud?.isServerlessEnabled);
+
+  let defaultPolicyPerType: PolicyConfig = useCloudConfig
+    ? getCloudPolicyConfig(factoryPolicy)
+    : getEndpointPolicyWithIntegrationConfig(factoryPolicy, config);
```

Comment:
@gergoabraham @opauloh 

Hey, tagging you both since I saw your names throughout this part of the code! I could use some help understanding this section. Let me explain what I’m trying to achieve:

In a Cloud deployment, when performing managed onboarding of Elastic Defend (there’s a recording in this PR’s description), I add an integration and go through the “multiPageLayout,” which creates an agent policy, enrolls an agent with it, and adds the Defend integration. In this flow, I would expect it to follow the `getCloudPolicyConfig()` path.

In this scenario, we create a package policy with `inputs: []`, which are then enriched in the `on create package policy` external callback. However, when `config?.type === 'cloud'`, since config is `undefined` (because it’s calculated based on inputs, which are empty in my case), it ends up going to `getEndpointPolicyWithIntegrationConfig`.

My question is: is this the expected behavior, or could this be an overlooked case? If it’s the latter, does my addition of the useCloudConfig check make sense?

Thanks in advance for your help!

Reply:
that's a nice catch! i'm not 100%, but i would say your solution makes sense, because as `config` is optional, there should be another way to define if we're on cloud or not. and we definitely need the `getCloudPolicyConfig()` in the cloud, so this is probably an overlooked case : )

nit:`cloud` is a mandatory parameter, so there's no need for optional chaining in line 54. just noticed this, because i was concerned that even `cloud` is optional, which would have mean that we don't have a definitive solution for decision making, but fortunately that's not the case

Reply:
Hey, it's been a while since I worked on this, so not sure if the requirements changed along the way, but IIRC the possible values of `config.type` are either `endpoint` or `cloud` and the correct value depends on what environment customers want to protect.

Endpoint:

![image](https://github.com/user-attachments/assets/a4e51254-a9e2-453c-be3c-c9bd885f0611)

Cloud:

![image](https://github.com/user-attachments/assets/cff74a3f-b586-4aa7-acf1-188365eab46c)

In the new `useCloudConfig` logic, it's assuming that when `config.type` is not set (i.e multiPageLayout deployment) it will be `cloud` on ESS environments and `endpoint` on Serverless environments (although it would not happen yet as `multiPageLayout` is only for ESS)

I'm not entirely sure of the assumption that someone installing Elastic Defend (through the multiPageLayout) will always want to deploy agents to protect their **Cloud Workload** environments instead of endpoints, because, currently in the Elastic Defend UI to add the integration, the default option is `endpoint`. 


Also, the stack environment (ESS or Serverless) doesn't matter as both environments can deploy both types of Policy, what matters is where the `elastic-agent` is going to be installed. So, Ideally, the type of environment should be determined by the user, but since that's not possible on the multiPageLayout UI, I think it is worth double checking with Product the desired behavior for this use case (making CloudPolicyConfig the default policy when deploying from `multiPageLayout`), and if confirmed I think we could omit the part that involves the environment, as it doesn't look like it really plays a role here, and move the condition to a helper with comments to describe best each option, and making the default option more intentional:

```js
const getDefaultPolicy = () : PolicyConfig  => {
    // Cloud workload protection
    if (config?.type === 'cloud') {
       return getCloudPolicyConfig(factoryPolicy);
    }
    // traditional endpoint protection
    if (config?.type === 'endpoint') {
        return getEndpointPolicyWithIntegrationConfig(factoryPolicy, config);
    }

    // In case config is not defined (i.e on managed cloud onboarding), defaults to Cloud Policy Config
    return getCloudPolicyConfig(factoryPolicy);	
}

let defaultPolicyPerType = getDefaultPolicy()
```

Reply:
@opauloh 

Thank you so much for the explanation, it cleared up a lot of questions I had, really appreciate it! I just have one more question - can I safely assume that a packagePolicy with `inputs: []`, which results in the config being `undefined`, will only occur in my scenario (managed onboarding)?

In the original code, anything that wasn’t `config?.type === 'cloud'` defaulted to `getEndpointPolicyWithIntegrationConfig`. With the current changes, we would effectively default to `getCloudPolicyConfig`. This makes me a bit nervous about potentially introducing regressions in areas I might not be aware of.

Reply:
>  can I safely assume that a packagePolicy with inputs: [], which results in the config being undefined, will only occur in my scenario (managed onboarding)?

That would be true when considering deploying through UI only, but we also support installation using API as we can see in the docs [here](https://www.elastic.co/guide/en/security/current/create-defend-policy-api.html#add-defend-integration). Also there's an active [issue](https://github.com/elastic/security-team/issues/6720) for improving it.


>  potentially introducing regressions in areas I might not be aware of.

Ideally, users installing Elastic Defend using the API method should also pass the `inputs` as instructed on the docs, but at the same time, there's nothing that prevents them from not passing it, which would result in saving the default policy as Cloud in the absence of inputs.

Since there's no information that we can properly rely on to identify if the integration was saved using the API or the managed cloud onboarding UI, I still think your solution makes sense, also the additional checking and comment makes clear that we want to use `getCloudPolicyConfig` for this use case.

Given the current scenario, it looks like the best option here is to proceed with your solution, I would also consider adding a `release_note:plugin_api_changes` label and a Plugin API Changes section explaining that when saving the Elastic Defend the API in the absence of the config inputs the integration Elastic Defend will now be saved with Cloud defaults.